content/2_Bronze/Intro_Sets.mdx

```diff
@@ -464,164 +466,50 @@ print("new dict:", d)
-"iterators, pointers and references referring to elements removed by the
-function are invalidated." So incrementing `it` after `it` has been erased from
-the map might not produce the intended result. If you're lucky, this will
-produce a segmentation fault. Unfortunately, sometimes (as in this case) the
-code will run without appearing to produce an error.
-
-If we compile using
-[`-D_GLIBCXX_DEBUG`](/general/debugging-lang/#-d_glibcxx_debug) and run the
-above, then
-
-```
-g++ -D_GLIBCXX_DEBUG whoops.cpp -o whoops && ./whoops
-```
-
-gives an error, as expected.
-
-```
-CURRENT KEY: 0
-CURRENT KEY: 1
-CURRENT KEY: 2
-/usr/local/Cellar/gcc/10.1.0/include/c++/10.1.0/debug/safe_iterator.h:328:
-In function:
-    __gnu_debug::_Safe_iterator<_Iterator, _Sequence, _Category>&
-    __gnu_debug::_Safe_iterator<_Iterator, _Sequence,
-    _Category>::operator++() [with _Iterator =
-    std::_Rb_tree_iterator<std::pair<const int, int> >; _Sequence =
-    std::__debug::map<int, int>; _Category = std::forward_iterator_tag]
-
-Error: attempt to increment a singular iterator.
-
-Objects involved in the operation:
-    iterator "this" @ 0x0x7ffee963c870 {
-      type = std::_Rb_tree_iterator<std::pair<int const, int> > (mutable iterator);
-      state = singular;
-      references sequence with type 'std::__debug::map<int, int, std::less<int>, std::allocator<std::pair<int const, int> > >' @ 0x0x7ffee963c8b0
-    }
-zsh: abort      ./whoops
-```
-
-Similarly, in
-[Java](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/Map.html),
-
-> If the map is modified while an iteration over the collection is in progress
-> (except through the iterator's own remove operation), the results of the
-> iteration are undefined.
-
-As suggested by
-[this](https://stackoverflow.com/questions/8234779/how-to-remove-from-a-map-while-iterating-it/39719450#39719450)
-StackOverflow post, the following code produces the intended results.
+One way to get around this is to just create a new map instead of removing from
+the old one.
 
 ```cpp
-void iterate_remove_ok() {
-	map<int,int> m; for (int i = 0; i < 10; ++i) m[i] = i;
-	int cnt = 0;
-	for (auto it = begin(m), next_it = it; it != end(m); it = next_it) {
-		++next_it;
-		cout << "CURRENT KEY: " << it->f << "\n";
-		++cnt;
-		if (cnt%3 == 0) {
-			m.erase(it);
-		}
-	}
-	cout << "REMAINING ENTRIES:\n";
-	for (pair<int,int> p: m)
-		cout << p.f << " " << p.s << "\n";
+map<int, int> m, M;
+for (int i = 0; i < 10; ++i)
+	m[i] = i;
+int cnt = 0;
+for (const auto &it : m) {
+	++cnt;
+	if (cnt % 3 != 0)
+		M[it.first] = it.second;
 }
+swap(m, M);
+cout << "Entries:" << endl;
+for (const auto &it : m)
+	cout << it.first << " " << it.second << endl;
 
 /*
-CURRENT KEY: 0
-CURRENT KEY: 1
-CURRENT KEY: 2
-CURRENT KEY: 3
-CURRENT KEY: 4
-CURRENT KEY: 5
-CURRENT KEY: 6
-CURRENT KEY: 7
-CURRENT KEY: 8
-CURRENT KEY: 9
-REMAINING ENTRIES:
+Remaining Entries:
```

Comment:
```suggestion
Entries:
```
i think this is what the code above says but i could be looking at the diff wrong